File: app/src/main/java/in/completecourse/questionbank/adapter/MoreAppsAdapter.java
package in.completecourse.questionbank.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import in.completecourse.questionbank.R;
import in.completecourse.questionbank.model.MoreAppsItem;

public class MoreAppsAdapter extends RecyclerView.Adapter<MoreAppsAdapter.ViewHolder> {
    private ArrayList<MoreAppsItem>  moreAppsItems;
    private Context mContext;

    public MoreAppsAdapter(Context mContext, ArrayList<MoreAppsItem> moreAppsItems) {
        this.moreAppsItems = moreAppsItems;
        this.mContext = mContext;
    }

    @NonNull
    @Override
    public MoreAppsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_more_apps, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MoreAppsAdapter.ViewHolder holder, int position) {
        MoreAppsItem item = moreAppsItems.get(position);

        Glide.with(mContext).load(item.getIcon_url()).into(holder.imgAppIcon);

        holder.tvAppName.setText(item.getApp_name());
        holder.tvAppDesc.setText(item.getApp_desc());
    }

    @Override
    public int getItemCount() {
        return moreAppsItems.size();
    }

     static class ViewHolder extends RecyclerView.ViewHolder {
        private TextView tvAppName, tvAppDesc;
        private ImageView imgAppIcon;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imgAppIcon = itemView.findViewById(R.id.img);
            tvAppName = itemView.findViewById(R.id.tv_title);
            tvAppDesc = itemView.findViewById(R.id.tv_subtitle);
        }
    }

    public void setItems(ArrayList<MoreAppsItem> items){
        this.moreAppsItems = items;
    }

}

File: app/src/main/java/in/completecourse/questionbank/adapter/MoreAppsAdapter.kt
package `in`.completecourse.questionbank.adapter

import `in`.completecourse.questionbank.R
import `in`.completecourse.questionbank.model.MoreAppsItem
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import java.util.*

class MoreAppsAdapter(mContext: Context, moreAppsItems: ArrayList<MoreAppsItem>) : RecyclerView.Adapter<MoreAppsAdapter.ViewHolder>() {
    private var moreAppsItems: ArrayList<MoreAppsItem>
    private val mContext: Context
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.item_more_apps, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item: MoreAppsItem = moreAppsItems[position]
        Glide.with(mContext).load(item.icon_url).into(holder.imgAppIcon)
        holder.tvAppName.text = item.app_name
        holder.tvAppDesc.text = item.app_desc
    }

    override fun getItemCount(): Int {
        return moreAppsItems.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvAppName: TextView = itemView.findViewById(R.id.tv_title)
        val tvAppDesc: TextView = itemView.findViewById(R.id.tv_subtitle)
        val imgAppIcon: ImageView = itemView.findViewById(R.id.img)

    }

    fun setItems(items: ArrayList<MoreAppsItem>) {
        moreAppsItems = items
    }

    init {
        this.moreAppsItems = moreAppsItems
        this.mContext = mContext
    }
}